zipline/pipeline/factors/technical.py

```diff
@@ -739,3 +742,59 @@ def compute(self, today, assets, out, close, k):
         out.middle = middle = nanmean(close, axis=0)
         out.upper = middle + difference
         out.lower = middle - difference
+
+
+class FastStochasticOscillator(CustomFactor):
+    """
+    Fast Stochastic Oscillator Indicator [%K, Momentum Indicator]
+    https://wiki.timetotrade.eu/Stochastic
+
+    This stochastic is considered volatile, and varies a lot when used in
+    market analysis. It is recommended to use the slow stochastic oscillator
+    or a moving average of the %K [%D].
+
+    **Default Inputs:** :data: `zipline.pipeline.data.USEquityPricing.close`
+                        :data: `zipline.pipeline.data.USEquityPricing.low`
+                        :data: `zipline.pipeline.data.USEquityPricing.high`
+
+    **Default Window Length:** 14
+
+    **Default D_period:** 3
+
+    Parameters
+    ----------
+    window_length : int > 0
+        Length of the lookback window over which to compute the stochastic
+        oscillator. Commonly called the K-period.
+    D_period : int
+        Period over which the K% is averaged (smoothed) over in units of days.
+
+    Returns
+    -------
+    out.K: %K oscillator
+    out.D: %D oscillator, which is %K averaged over the last D_period days
+    """
+    params = ('D_period', )
+    inputs = (USEquityPricing.close, USEquityPricing.low, USEquityPricing.high)
+    window_length = 14
+    outputs = 'K', 'D'
+
+    def compute(self, today, assets, out, closes, lows, highs, D_period=3):
```

Comment:
This default won't ever actually be used the way the the `params` machinery works right now.  I'd probably just make it required.